src/app/static/src/app/components/modelCalibrate/CalibrationResults.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+    <div>
+        <hr />
+        <h3 v-translate="'calibrateResultsHeader'"></h3>
+        <p class="text-muted" v-translate="'calibrateResultsDesc'"></p>
+        <bar-chart-with-filters
+            :chart-data="chartData"
+            :filter-config="filterConfig"
+            :indicators="indicators"
+            :selections="selections"
+            :formatFunction="formatBarchartValue"
+            :x-axis-config="config"
+            :disaggregate-by-config="config"
+            @update="updateCalibratePlotSelections({ payload: $event })"
+        ></bar-chart-with-filters>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import i18next from "i18next";
+    import { Language, Translations } from "../../store/translations/locales";
+    import { BarchartIndicator, Filter } from "@reside-ic/vue-charts/src/bar/types";
+    import {
+        BarChartWithFilters,
+        FilterConfig,
+        FilterOption,
+    } from "@reside-ic/vue-charts";
+    import {
+        mapGettersByNames,
+        mapMutationByName,
+        mapMutationsByNames,
+        mapStateProp,
+    } from "../../utils";
+    import {
+        PlottingSelectionsState,
+        BarchartSelections,
+        UnadjustedBarchartSelections,
+    } from "../../store/plottingSelections/plottingSelections";
+    import { ModelCalibrateState } from "../../store/modelCalibrate/modelCalibrate";
+    import { formatOutput } from "../plots/utils";
+    import { RootState } from "../../root";
+    import { PayloadWithType, BarchartConfig } from "../../types";
+
+    const namespace = "modelCalibrate";
+
+    interface Data {
+        config: BarchartConfig
+    }
+
+    interface Methods {
+        updateCalibratePlotSelections: (data: {
+            payload: BarchartSelections;
+        }) => void;
+        formatBarchartValue: (
+            value: string | number,
+            indicator: BarchartIndicator
+        ) => string;
+    }
+
+    interface Computed {
+        calibratePlotDefaultSelections: UnadjustedBarchartSelections;
+        chartData: any;
+        filterConfig: FilterConfig;
+        filters: Filter[];
+        selections: BarchartSelections;
+        indicators: BarchartIndicator[];
+        currentLanguage: Language;
+    }
+
+    export default Vue.extend<unknown, Methods, Computed, unknown>({
+        name: "CalibrationResults",
+        components: {
+            BarChartWithFilters,
+        },
+        data() {
+            return {
+                config: {
```

Comment:
I don't know how to get the vue template to read a private variable without arbitrarily assigning it to a data prop or returning it from a method. I tried assigning it to `this` using the created hook but then ts complains that it doesn't exist.

Reply:
Apologies, I thought you could access consts from the template, evidently not. In that case, I still think it doesn't belong in `data`, since that makes it a reactive object unnecessarily. It would be better to return it from a method, or to be honest I'd probably just hard-code it into the prop in the template - it does mean repeating it, but only once, and it's only little.